kernel/src/main/java/org/kframework/parser/concrete2kore/generator/RuleGrammarGenerator.java

```diff
@@ -200,7 +200,7 @@ public static ParseInModule getCombinedGrammar(Module mod, boolean strict, boole
         if (mod.importedModuleNames().contains(AUTO_CASTS)) { // create the diamond
             Set<Sentence> temp;
             for (Sort srt : iterable(mod.definedSorts())) {
-                if (!isParserSort(srt)) {
+                if (!isParserSort(srt) || mod.subsorts().directlyLessThan(Sorts.KVariable(), Sorts.RuleTag())) {
```

Comment:
I believe this should be directlyLessThan(Sorts.KVariable(), srt), right?

Reply:
Haha, good catch. Fixed.